dissect/target/filesystem.py

```diff
@@ -870,7 +870,7 @@ def lattr(self) -> Any:
         raise TypeError(f"lattr is not allowed on VirtualDirectory: {self.path}")
 
     def add(self, name: str, entry: FilesystemEntry) -> None:
-        """Add an entry to this :class:`VirtualDirectory`."""
+        """Add an entry to this :class:`VirtualDirectory`. Overrides already existing entries with ``name``."""
```

Comment:
Maybe we can add a `clobber: bool` argument to `add`? And it could either raise a `KeyError` or something on conflict, or at least log it.

If possible, I'd like to solve this in `VirtualFilesystem` as much as possible.